source/language/pulses.rst

```diff
@@ -92,3 +79,217 @@ circuit. Most of the time symbols in the ``defcal`` table will also have
+   // center is the mean of pulse
+   // sigma is the standard deviation of pulse
+   sech(length:l, complex[float[32]]:amp, length:center, length:sigma)
+
+   // amp is pulse amplitude at center
+   // center is the mean of pulse
+   // square_width is the width of the square pulse component
+   // sigma is the standard deviation of pulse
+   gaussian_square(length:l, complex[float[32]]:amp, length:center, length:square_width, length:sigma)
+
+   // amp is pulse amplitude at center
+   // center is the mean of pulse
+   // sigma is the standard deviation of pulse
+   // beta is the Y correction amplitude, see the DRAG paper
+   drag(length:l, complex[float[32]]:amp, length:center, length:sigma, float[32]:beta)
+
+Shift Phase Instruction
+-----------------------
+
+A ``shift_phase`` instruction schedules a phase advance for all the following
+pulses on that channel. This is equivalent to multiplying each pulse by
+:math:`e^{-i*\theta}`, where theta is the phase change in radians.
+
+The ``shift_phase`` instruction takes two parameters, a channel and the
+requested phase change of type angle. The exact precision of the implemented
+phase change will vary depending on hardware support.
+
+.. code-block:: none
+
+   // Shift phase of qubit 0 by pi/4, eg. an rz gate with angle -pi/4
+   shift_phase drive(%0), pi/4;
+
+   // Define a calibration for the rz gate on all physical qubits
+   defcal rz(angle[20]:theta) %q {
+     shift_phase drive(%q), -theta;
+   }
+
+Shift Frequency Instruction
+---------------------------
+
+A ``shift_freq`` instruction schedules a frequency advance for all the following
+pulses on that channel. This is useful for defining spectroscopy experiments.
+
+The ``shift_freq`` instruction takes two parameters, a channel and the
+requested frequency change of type float in units of GHz.
+
+Here's an example qubit spectroscopy experiment. Note that the starting
+frequency will be defined somewhere outside OpenQASM.
+
+.. code-block:: none
+
+   qubit q;
+
+   const shots = 1000;
+   const dfreq = 0.001; # 1MHz per point
+   const points = 50; # Sweep over 50MHz
+
+   complex[float[32]] iq, average;
+   complex[float[32]] output[points];
+
+   for p in [0 : points-1] {
+     average = 0;
+     for i in [0 : shots-1] {
+       // Assumes suitable calibrations for reset, x, and measure_iq
+       reset q;
+       x q;
+       measure_iq q -> iq;
+
+       average = (average * i + iq) / (i + 1);
+     }
+     shift q;
+     output[p] = average;
+   }
+
+   defcal shift %q {
+     shift_freq drive(%q) dfreq;
+   }
+
+Capture Instruction
+-------------------
+
+Acquisition is scheduled by a ``capture`` instruction.
+
+The ``capture`` instruction takes two parameters, a channel and the filter to
+apply to the returned signal. The length of the filter in dt will determine how
+long the capture channel is open. The ``capture`` instruction returns the dot
+product of the measured IQ values and the filter. (Note that the filter is
```

Comment:
I'll update "other contexts" to what I really meant which was "some literature uses the term kernel for what we refer to as a filter, since kernel in OpenQASM is already used to refer to a custom classical function"